### ISSUE TITLE
[BUG] host.py: Modifying host with same folder (e.g. just change attribute or ipaddress) fails

### ISSUE THREAD

**Issue by muehlings** [tribe29/ansible-collection-tribe29.checkmk#99]
Seems related to these issues:
https://github.com/tribe29/ansible-collection-tribe29.checkmk/issues/58
https://github.com/tribe29/ansible-collection-tribe29.checkmk/issues/42

Playbook:
1. First run (host never existed before) is working:
```
  - name: "Create a host with IP address."
    tribe29.checkmk.host:
      server_url: "https://localhost"
      site: "site"
      automation_user: "automation"
      automation_secret: "$SECRET"
      host_name: "my_host_with_ip"
      attributes:
        alias: "My Host"
        # ip_address from example is wrong:
        ipaddress: "127.0.0.1"
        tag_custom: "own_custom_tag"
      folder: "/"
      state: "present"
```

2. Second run with new IP fails:
```
  - name: "Create a host with IP address."
    tribe29.checkmk.host:
      server_url: "https://localhost"
      site: "site"
      automation_user: "automation"
      automation_secret: "$SECRET"
      host_name: "my_host_with_ip"
      attributes:
        alias: "My Host"
        # ip_address from example is wrong:
        ipaddress: "127.0.0.2"
        tag_custom: "own_custom_tag"
      folder: "/"
      state: "present"
```

Result:
```
TASK [Create a host with IP address.] ******************************************
fatal: [site]: FAILED! => {"changed": false, "msg": "Error calling API. HTTP code 400. Details: b'{\"title\": \"Invalid move action\", \"status\": 400, \"detail\": \"The host is already part of the specified target folder\"}', "}
```

Desired result:
```
TASK [Create a host with IP address.] ******************************************
changed: [site]
```

I found the position where "move_host()" is triggered and analyzed it:

"/" is stripped from current_folder...
https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/d94e27fe8bd598de78fe4df3c75529d7bdbc21af/plugins/modules/host.py#L284
https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/d94e27fe8bd598de78fe4df3c75529d7bdbc21af/plugins/modules/host.py#L285

```
        if current_folder.endswith("/"):
            current_folder = current_folder.rstrip("/")
```

...but not from:
https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/d94e27fe8bd598de78fe4df3c75529d7bdbc21af/plugins/modules/host.py#L287

I fixed it temporary with module.params["folder"]**.rstrip("/")**

```
        if current_folder != module.params["folder"].rstrip("/"): # MU: added .rstrip("/")
            move_host(module, base_url, headers)
            msg_tokens.append("Host was moved.")
```

I have not checked for side-effects.

Side-note:
The example has "ip_address" as attribute, but it's actually "ipaddress".

**Comment by muehlings** [comment]
I have tried the devel-branch (https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/8928a1df37701c863752fce5692b5e17c6ae2b5c/plugins/modules/host.py) and my example tasks are working now. 
- Changing folder and/or ip of the host is working. 
- The state (absent/present) is working.

But the role is missing a line for the state: https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/8928a1df37701c863752fce5692b5e17c6ae2b5c/roles/agent/tasks/main.yml#L34-L44

Adding `state: "{{ checkmk_agent_state | default(omit) }}"` after https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/8928a1df37701c863752fce5692b5e17c6ae2b5c/roles/agent/tasks/main.yml#L43 seems to fix it:
`changed: [my_host_with_ip -> localhost] => {"changed": true, "failed_when_result": false, "msg": "Host deleted."}`

My inventory now looks like:

```
my_site: # = Sitename
  hosts:
    host_with_ip:
      ansible_host: 127.0.0.1
      checkmk_agent_host_attributes:
        ipaddress: '{{ansible_host}}'
        tag_custom: "own_custom_tag"
      checkmk_agent_folder: test
      state: present
    my_host_with_ip:
      ansible_host: 127.0.0.2
      checkmk_agent_host_attributes:
        ipaddress: '{{ansible_host}}'
        tag_custom: "own_custom_tag"
      checkmk_agent_folder: test
      state: absent
  vars:
    checkmk_agent_add_host: 'true'
    checkmk_agent_edition: cme
    checkmk_agent_host_ip: '{{ ansible_host }}'
    checkmk_agent_host_name: '{{ inventory_hostname }}'
    checkmk_agent_pass: "$SECRET"
    checkmk_agent_protocol: https
    checkmk_agent_server: localhost
    checkmk_agent_site: my_site
    checkmk_agent_user: automation
    checkmk_agent_version: 2.0.0p23
```

My playbook now looks like:
```
---
- hosts: my_site # = Sitename
  connection: local

  roles:
  - tribe29.checkmk.agent
```

**Comment by robin-tribe29** [comment]
@lgetwan can you take a look at the plugin, I cannot seem to get my head around it.

**Comment by lgetwan** [comment]
Fixed with PR #147
